### ISSUE TITLE
Flutter Test should support multiple reporters

### ISSUE THREAD

**Issue by danilofuchs** [flutter/flutter#69425]
## Use case

When running `flutter test` on a CI, output is very verbose and hard to follow:

![image](https://user-images.githubusercontent.com/17657014/97740642-56f6f400-1ac0-11eb-8118-a0bf43f0629b.png)
(this goes on for 200+ lines)

The current alternative is to output to json with `flutter test --machine`. This is even more verbose, but can be converted to JUnit XML so we can use a specialized test visualization tools, such as Bitrise Test Reports.

This does the job, but requires a lot of configuration and there is currently no way of having two reporters at the same time.
If I want both CLI output and JSON, I need to run tests twice.
I want to be able to output to any format while still having a console output (customizable as well).

## Proposal

Flutter should have a `--reporter` CLI option, inspired by [Mocha Reporters](https://mochajs.org/#reporters) or [Jest Reporters.](https://jestjs.io/docs/en/cli#--reporters)

```bash
# Current output if not in CI environment
flutter test

# Simple, CI friendly output
flutter test --reporter spec

# Minimal output with number of passes AND json output to test_results.json file
flutter test --reporter min --reporter json --reporter-option output=test_results.json

# JUnit XML report out of the box
flutter test --reporter xunit  --reporter-option output=test_results.xml
```

By default Flutter will use the current reporter. If env `CI=true`, flutter will use `spec` automatically.

This is an idea we can take from the very well established and mature JS ecosystem.

**Comment by ogezue** [comment]
is there something on the market that can be used instead?

**Comment by nico-famedly** [comment]
Dart supports the --file-reporter option: https://github.com/dart-lang/test/blob/master/pkgs/test/doc/json_reporter.md#usage

It would be nice if flutter supported that too.

**Comment by bartekpacia** [comment]
`flutter test` does have a `--reporter` argument, which accepts `compact`, `expanded`, and `json`. Is this what you're looking for?

**Comment by danilofuchs** [comment]
> `flutter test` does have a `--reporter` argument, which accepts `compact`, `expanded`, and `json`. Is this what you're looking for?

The main goal is outputting to more than one reporter at the same time. So JSON + expanded in a single run.

Optionally, allow custom reporters, for developing plugins for jUnit XML, ...
